lib/boombox.ex

```diff
@@ -220,6 +246,68 @@ defmodule Boombox do
       {:mp4, location, _opts} when is_binary(location) and direction == :output ->
         value
 
+      {:h264, location} when is_binary(location) and direction == :input ->
+        parse_endpoint_opt!(:input, {:h264, location, []})
+
+      {:h264, location, opts} when is_binary(location) and direction == :input ->
+        if Keyword.keyword?(opts),
+          do:
+            {:h264, location,
+             transport: resolve_transport(location, opts), framerate: opts[:framerate] || {30, 1}}
+
+      {:h264, location} when is_binary(location) and direction == :output ->
+        {:h264, location}
+
+      {:aac, location} when is_binary(location) and direction == :input ->
+        parse_endpoint_opt!(:input, {:aac, location, []})
+
+      {:aac, location, opts} when is_binary(location) and direction == :input ->
+        if Keyword.keyword?(opts),
+          do: {:aac, location, transport: resolve_transport(location, opts)}
+
+      {:aac, location} when is_binary(location) and direction == :output ->
+        {:aac, location}
+
+      {:wav, location} when is_binary(location) and direction == :input ->
+        parse_endpoint_opt!(:input, {:wav, location, []})
+
+      {:wav, location, opts} when is_binary(location) and direction == :input ->
+        if Keyword.keyword?(opts),
+          do: {:wav, location, transport: resolve_transport(location, opts)}
+
+      {:wav, location} when is_binary(location) and direction == :output ->
+        {:wav, location}
+
+      {:mp3, location} when is_binary(location) and direction == :input ->
+        parse_endpoint_opt!(:input, {:mp3, location, []})
+
+      {:mp3, location, opts} when is_binary(location) and direction == :input ->
+        if Keyword.keyword?(opts),
+          do: {:mp3, location, transport: resolve_transport(location, opts)}
+
+      {:mp3, location} when is_binary(location) and direction == :output ->
+        {:mp3, location}
+
+      {:ivf, location} when is_binary(location) and direction == :input ->
+        parse_endpoint_opt!(:input, {:ivf, location, []})
+
+      {:ivf, location, opts} when is_binary(location) and direction == :input ->
+        if Keyword.keyword?(opts),
+          do: {:ivf, location, transport: resolve_transport(location, opts)}
+
+      {:ivf, location} when is_binary(location) and direction == :output ->
+        {:ivf, location}
+
+      {:ogg, location} when is_binary(location) and direction == :input ->
+        parse_endpoint_opt!(:input, {:ogg, location, []})
+
+      {:ogg, location, opts} when is_binary(location) and direction == :input ->
+        if Keyword.keyword?(opts),
+          do: {:ogg, location, transport: resolve_transport(location, opts)}
+
+      {:ogg, location} when is_binary(location) and direction == :output ->
+        {:ogg, location}
+
```

Comment:
Some duplication here too, I think it should be possible to handle the 'common cases' in one place

Reply:
done